src/app/WriteHandler.cpp

```diff
@@ -567,24 +571,33 @@ Status WriteHandler::ProcessWriteRequest(System::PacketBufferHandle && aPayload,
 #if CHIP_CONFIG_IM_PRETTY_PRINT
     writeRequestParser.PrettyPrint();
 #endif
-    err = writeRequestParser.GetSuppressResponse(&mSuppressResponse);
+    bool boolValue;
+
+    boolValue = mStateFlags.Has(StateBits::kSuppressResponse);
+    err       = writeRequestParser.GetSuppressResponse(&boolValue);
+    mStateFlags.Set(StateBits::kSuppressResponse, boolValue);
```

Comment:
Now that we have broken up the GetSuppressResponse and the Set into separate lines of code, the Set should happen below after the SuccessOrExit(err), no?

Similar for the other bits below.

Reply:
This is what I had initially, however that is not what was happening in the previous code: in the previous code, the bool setting from the `Get` was happening regardless of the return code of success or error.

So to replicate a `GetFoo(mMember)`, the exact equivalent is `boolValue = flags.Has(Member); Get(boolValue); flags.Set(Member, boolValue)`

Reply:
I understand that, but the previous code was sort of relying on things it would be better to not rely on to save a few lines of code, and we should not keep relying on them, since we have those extra lines of code anyway.

Reply:
#34992 makes this change.